src/methods/extract.jl

```diff
@@ -64,12 +64,14 @@ function extract(x::RasterStackOrArray, data;
         NamedTuple{keys}
     end
     names = NamedTuple{names}(names)
-    if Tables.istable(data)
+    if Tables.istable(data) && !(data isa AbstractVector{<:GeoInterface.NamedTuplePoint})
```

Comment:
```suggestion
    if !(data isa AbstractVector{<:GeoInterface.NamedTuplePoint}) && Tables.istable(data)
```
The cheap one goes first. In many cases compiler wont be able to prove that `istable` has no side effects so it will have to run it.